sentry/src/main/java/io/sentry/SentryEnvelopeItem.java

```diff
@@ -148,6 +150,49 @@ public static SentryEnvelopeItem fromUserFeedback(
     return new SentryEnvelopeItem(itemHeader, cachedItem::getBytes);
   }
 
+  public static SentryEnvelopeItem fromAttachment(
+      @NotNull ILogger logger, final @NotNull Attachment attachment) {
+
+    final CachedItem cachedItem =
+        new CachedItem(
+            () -> {
+              if (attachment.getBytes() != null) {
+                return attachment.getBytes();
+              } else if (attachment.getPath() != null) {
+
+                try (FileInputStream fileInputStream = new FileInputStream(attachment.getPath());
+                    BufferedInputStream inputStream = new BufferedInputStream(fileInputStream);
+                    ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+
+                  byte[] bytes = new byte[1024];
+                  int length;
+                  int offset = 0;
+                  while ((length = inputStream.read(bytes)) != -1) {
+                    outputStream.write(bytes, offset, length);
+                  }
+                  return outputStream.toByteArray();
+                } catch (IOException | SecurityException exception) {
+                  logger.log(
+                      SentryLevel.ERROR,
+                      exception,
+                      "Serializing attachment %s failed.",
+                      attachment.getFilename());
+                }
+              }
+
+              return new byte[0];
+            });
+
+    SentryEnvelopeItemHeader itemHeader =
+        new SentryEnvelopeItemHeader(
+            SentryItemType.Attachment,
+            () -> cachedItem.getBytes().length,
+            attachment.getContentType(),
+            attachment.getFilename());
+
+    return new SentryEnvelopeItem(itemHeader, cachedItem::getBytes);
```

Comment:
H: `::` can cause issues on Android.
not all AGP versions do the method references translation properly

See: DefaultAndroidEventProcessor
there's a comment about it.

Reply:
Thanks, didn't know that. Again a linter would be nice.